webforces/views.py

```diff
@@ -81,6 +86,57 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class StoreView(MainPageView):
+    template_name = "store.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        core = Core()
+        status, algorithms_list = core.db.getAllAlgs()
+
+        status, user = core.db.getUserByLogin(self.request.user.username)
+        if status != DBStatus.s_ok:
+            messages.error(self.request, "Internal error: can not find current user!")
+        else:
+            context["algorithms_list"] = algorithms_list
+        return context
+
+
+class AddAlg(FormView):
+    template_name = "add_alg.html"
+    form_class = NewAlgForm
+    success_url = '.'
```

Comment:
I think it'll be more convenient to redirect user to `/store` after algorithm addition